File: Programmers/Level2/문자열_압축.py
def solution(s):
    n = len(s)
    answer = n
    for i in range(1, n+1):
        front = s[:i]
        cnt = 1
        idx = i
        li = []
        for idx in range(i, n, i):
            curStr = s[idx:idx+i]
            if front == curStr:
                cnt += 1
            else:
                tmp = (str(cnt) + front) if cnt > 1 else front
                li.append(tmp)
                cnt = 1
                front = curStr
        li.append((str(cnt) + front) if cnt > 1 else front)
        res = "".join(li)
        answer = min(answer, len(res))
    return answer

def main():
    s = "xababcdcdababcdcd"
    print(solution(s))
if __name__ == '__main__':
    main()

File: Programmers/Level2/문자열_압축.java
//문자열 압축 2020KAKAO BLIND RECRUITMENT 20211107
import static java.lang.Math.*;

class Solution {
    public int solution(String s) {
        int n = s.length();
        int answer = n;
        char[] arr = s.toCharArray();
        for(int k = 1; k <= n; k++){
            int offset = 0, cnt = 1;
            
            String prev = new String(arr, offset, k);
            StringBuilder sb = new StringBuilder();
            offset += k;
            
            while(offset + k <= n){
                String str = new String(arr, offset, k);
                if(prev.equals(str)){
                    cnt += 1;
                }else{
                    if(cnt != 1) sb.append(cnt);
                    sb.append(prev);
                    cnt = 1;
                }
                prev = str;
                offset += k;
            }
            if(cnt != 1) sb.append(cnt);
            sb.append(prev);
            
            if(offset != n){
                sb.append(new String(arr, offset, n - offset));
            }
            answer = min(answer, sb.toString().length());
        }
    
        return answer;
    }
}